core/src/main/java/org/apache/accumulo/core/fate/FateExecutorMetrics.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.accumulo.core.fate;
+
+import java.util.Set;
+import java.util.concurrent.TransferQueue;
+
+import org.apache.accumulo.core.metrics.Metric;
+import org.apache.accumulo.core.metrics.MetricsProducer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.micrometer.core.instrument.Gauge;
+import io.micrometer.core.instrument.MeterRegistry;
+
+public class FateExecutorMetrics<T> implements MetricsProducer {
+  private static final Logger log = LoggerFactory.getLogger(FateExecutorMetrics.class.getName());
+  private final FateInstanceType type;
+  private final String operatesOn;
+  private final Set<FateExecutor<T>.TransactionRunner> runningTxRunners;
+  private final TransferQueue<FateId> workQueue;
+  private MeterRegistry registry;
+  public static final String INSTANCE_TYPE_TAG_KEY = "instanceType";
+  public static final String OPS_ASSIGNED_TAG_KEY = "ops.assigned";
+
+  protected FateExecutorMetrics(FateInstanceType type, String operatesOn,
+      Set<FateExecutor<T>.TransactionRunner> runningTxRunners, TransferQueue<FateId> workQueue) {
+    this.type = type;
+    this.operatesOn = operatesOn;
+    this.runningTxRunners = runningTxRunners;
+    this.workQueue = workQueue;
+  }
+
+  @Override
+  public void registerMetrics(MeterRegistry registry) {
+    Gauge.builder(Metric.FATE_OPS_THREADS_TOTAL.getName(), runningTxRunners::size)
+        .description(Metric.FATE_OPS_THREADS_TOTAL.getDescription())
+        .tag(INSTANCE_TYPE_TAG_KEY, type.name().toLowerCase()).tag(OPS_ASSIGNED_TAG_KEY, operatesOn)
```

Comment:
Seems like operatesOn could be a really long string.  Wonder if that could cause problems for any metrics systems.  Also wondering if we could shorten it in anyway, like if it is operating on all operations with the TABLE prefix then replace that with `TABLE_*`.

Reply:
One options may be to give the set user defined names in config.  Then those names could be used in metrics and maybe other places.

Reply:
I'm not sure I follow

Reply:
If we had a config value like the following for `manager.fate.threadpool.size`  then we could use `table_ops` and `system_ops` for logging and metrics instead of all of the values in the set.    `table_ops` and `system_ops` are user defined names.

```json
{"table_ops": {"operations":"TABLE_...", "threads":4}, "system_ops":{"operations":"COMMIT_COMPACTION,SYSTEM_SPLIT ","threads":4}} 
```

Reply:
Is there any part of the metric we could use that wouldn't have length limitations? For example, could we include the ops in the metric description? Maybe not, I think it might be a limitation on the metric as a whole...

Reply:
Wait I think this is exactly what you were saying...

Reply:
yeah definitely not a change for this PR, was just something I noticed while looking at this PR.